packages/rum-core/src/performance-monitoring/metrics.js

```diff
@@ -174,18 +175,36 @@ export function calculateTotalBlockingTime(longtaskEntries) {
 }
 
 /**
- * Calculates Cumulative Layout Shift using
+ * Calculates Cumulative Layout Shift through the max-session-gap1s-limit5s technique using
  * 'layout-shift' entries captured through PerformanceObserver.
  * https://developer.mozilla.org/en-US/docs/Web/API/LayoutShift
- * https://web.dev/cls/
+ * https://web.dev/evolving-cls/
+ *
+ * We use clsState to keep track of previous layout shift events since the used technique requires
+ * to have them into account to perform the proper calculation
  */
-export function calculateCumulativeLayoutShift(clsEntries) {
+export function calculateCumulativeLayoutShift(clsState, clsEntries) {
   clsEntries.forEach(entry => {
     // Only add the layout shift events without recent user input
     if (!entry.hadRecentInput && entry.value) {
-      metrics.cls += entry.value
+      const shouldCreateNewSession =
+        entry.startTime - clsState.firstEventTime > 5000 ||
+        entry.startTime - clsState.previousEventTime > 1000
+      if (shouldCreateNewSession) {
+        clsState.firstEventTime = entry.startTime
+        clsState.currentSessionScore = 0
+      }
+
+      clsState.previousEventTime = entry.startTime
+      clsState.currentSessionScore += entry.value
+      clsState.maxScore = Math.max(
```

Comment:
you could also get rid of maxScore, instead do metrics.cls.score to reduce the no of fields. Just for reduce bytes, nothing wrong with the current logic :)